rr_common/src/planner/CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
-add_executable(planner planner.cpp)
-target_link_libraries(planner ${catkin_LIBRARIES})
-add_dependencies(planner rr_platform_generate_messages_cpp)
+add_library(planner planner.cpp planner.h)
+# target_
```

Comment:
Why does this work without linking any libraries?

Reply:
A certain amount of linking happens when building the library, but libraries are allowed to be built without being fully linked. So, yes, when the final executable is linked, it will find the missing library dependencies in `catkin_LIBRARIES`. 

Side note, these target names should be changed to be more descriptive. (and/or just get rid of the intermediate library target?)